src/main/java/uk/gov/hmcts/reform/security/keyvault/credential/AccessTokenKeyVaultCredential.java

```diff
@@ -0,0 +1,57 @@
+package uk.gov.hmcts.reform.security.keyvault.credential;
+
+import com.microsoft.azure.AzureEnvironment;
+import com.microsoft.azure.credentials.AzureTokenCredentials;
+import org.apache.http.HttpResponse;
+import org.apache.http.client.HttpClient;
+import org.apache.http.client.ServiceUnavailableRetryStrategy;
+import org.apache.http.client.methods.HttpUriRequest;
+import org.apache.http.client.methods.RequestBuilder;
+import org.apache.http.impl.client.HttpClientBuilder;
+import org.apache.http.protocol.HttpContext;
+
+import java.io.IOException;
+
+import static uk.gov.hmcts.reform.security.keyvault.KeyVaultConfig.VAULT_ERROR_MAX_RETRIES;
+import static uk.gov.hmcts.reform.security.keyvault.KeyVaultConfig.VAULT_ERROR_RETRY_INTERVAL_MILLIS;
+
+public class AccessTokenKeyVaultCredential extends AzureTokenCredentials {
+
+    private static final String METADATA_HEADER = "Metadata";
+
+    private final String tokenEndpoint;
+
+    private static final TokenResponseHandler tokenResponseHandler = new TokenResponseHandler();
+
+    private static final HttpClient httpClient = HttpClientBuilder.create().setServiceUnavailableRetryStrategy(
+        new ServiceUnavailableRetryStrategy() {
+            @Override
+            public boolean retryRequest(final HttpResponse response,
+                                        final int executionCount, final HttpContext context) {
+                int statusCode = response.getStatusLine().getStatusCode();
+                return statusCode == 500
+                    && executionCount < Integer.valueOf(System.getProperty(VAULT_ERROR_MAX_RETRIES));
```

Comment:
I know this is being discussed in the other PR, but I'm really not a fan of using System.getProperty here

Reply:
Yep, still working out best approach.